crates/proto/src/error.rs

```diff
@@ -237,6 +238,15 @@ pub enum ProtoErrorKind {
     #[error("request timed out")]
     Timeout,
 
+    /// Tsig key verification failed
+    #[error("Tsig key wrong key error")]
+    TsigWrongKey,
+
+    /// Tsig unsupported mac algorithm
+    /// Supported algorithm documented in `TsigAlgorithm::supported` function.
+    #[error("Tsig unsupported mac algorithm")]
+    TsigUnsupportedMacAlgorithm(Name),
```

Comment:
Rather than storing a `Name` here, would it be better to instead use the Algorithm? would save a little allocation space until fmt is called from Display or Debug formatting...